### PR TITLE
popular screen sizes added. FIX #9

### LINKED ISSUE #9: Implement all popular mobile device sizes and platform support
we have only one size - iphone6, it should be possible to choose any popular device android or ios
also there should be platform support in the editor, so we can show differences in views for ios and android https://github.com/status-im/status-fiddle/blob/master/src/cljs/status_im/utils/platform.cljs

### PR DESCRIPTION
**Pull request by ghost**
Fixes #9

status: ready

### CODE REVIEW

**Comment by flexsurfer** [comment]
thanks @poiga works good, two moments
1) is it possible to use droplist instead buttons and positioning above screen?
image
2)looks like for android devices width and height reversed

**Comment by poiga** [comment]
@flexsurfer fixed those. please check!

**Comment by flexsurfer** [comment]
@poiga doesn't work, looks like you need to use `select onchange`

**Comment by ghost** [comment]
@flexsurfer Should be good now! Please check! 
Strange - it worked on Firefox even before but now I test in chrome as well!

File: src/cljs/status_fiddle/views.cljs [hunk 1]
```diff
@@ -178,7 +187,10 @@
     [react/view {:style {:flex 1}}
      [code-mirror]]
     [react/view {:style {:margin-left 50}}
-     [react/text "iPhone 6"]
-     [react/view {:style {:width 375 :height 667 :border-width 1 :border-color :blue}}
+     [phone-chooser]
+     [react/view {:style {
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/views.cljs]: extra line

File: src/cljs/status_fiddle/views.cljs [hunk 2]
```diff
@@ -151,6 +152,14 @@
     [color-panel styles/separator-color "styles/separator-color"]]
    [color-label]])
 
+(defn phone-chooser []
+  [:select
+   {:on-change #(let [selected (cljs.reader/read-string (.. % -target -value))]
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/views.cljs]: are you sure we should use `cljs.reader` here ?

File: src/cljs/status_fiddle/views.cljs [hunk 3]
```diff
@@ -2,6 +2,7 @@
   (:require [re-frame.core :as re-frame]
             [reagent.core :as reagent]
             [cljs.js :refer [eval-str empty-state js-eval]]
+            [status-fiddle.supported-devices :refer [devices]]
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/views.cljs]: please use `:as` instead `:refer`

File: src/cljs/status_fiddle/supported-devices.cljs [hunk 4]
```diff
@@ -0,0 +1,14 @@
+(ns status-fiddle.supported-devices)
+
+(def devices [
+   {:phone-name "iPhone 4S/4" :screen-width 320 :screen-height 440}
+   {:phone-name "iPhone 5S/5" :screen-width 320 :screen-height 568}
+   {:phone-name "iPhone 6" :screen-width 375 :screen-height 667}
```
Comment (flexsurfer) [inline src/cljs/status_fiddle/supported-devices.cljs]: can we select iPhone 6 by default?

**Review by flexsurfer — CHANGES_REQUESTED** [review]

**Comment by flexsurfer** [comment]
@poiga and what about platform support in the editor, so we can show differences in views for ios and android https://github.com/status-im/status-fiddle/blob/master/src/cljs/status_im/utils/platform.cljs ?

**Comment by flexsurfer** [comment]
hey @audriu i can see your changes. about platform support let's add just current-os, and change it depends on current device, and in platform read this variable, and add this ns to compile, i don't think we need this ns how it works in status-react because we will need two versions of fiddle for ios and android then, what do you think?

**Comment by ghost** [comment]
@flexsurfer I think I got you. I will make changes and I let's see...

**Comment by ghost** [comment]
@flexsurfer I think that instead 2 versions of fiddle we should build a preprocessor for the `android1 and `ios` tags?

**Comment by flexsurfer** [comment]
@audriu don't think i got your idea, let's implement simple version, so when you change device in listbox we should not only change size but also recompile if device os was changed, and add `platform` ns alias, in editor we should be able to use
```clojure
[react/test {style {if (platform/ios?} 2 3}
```
`platform/platform` should return "ios" or "android" string
`platform/ios?` false/true
`platform/android?` false/true

**Comment by ghost** [comment]
@flexsurfer please check my last commit! Is it similar to your idea?

**Comment by ghost** [comment]
@flexsurfer and please reload the default template! I added `(if (platform/ios?) :white :red)`

**Comment by flexsurfer** [comment]
@audriu yes, but this will not work, because user ns knows nothing about re-frame we need to use atom and reset! it manually. 
also i made some changes please rebase

**Comment by ghost** [comment]
@flexsurfer I think it is done now!

**Comment by flexsurfer** [comment]
hey @audriu this will not work too, just wondering do you make changes without running code? :) so i think this will be simplest solution
```clojure
"(ns cljs.platform)
                                  (def platform \"" os "\")
                                  (def android? " (= os "android") ")"
                                 "(def ios? " (= os "ios") ")"
```
i'll merge your PR and make changes, this will be faster, thanks

**Comment by ghost** [comment]
@flexsurfer it did work! seriously it did work with `(if (platform/ios?) :white :red)`. May be you wanted some different syntax, but it did work just like that.

**Comment by ghost** [comment]
@flexsurfer please try with snippet:

`[react/text {:style {:color(if (platform/ios?) :green :red)}}"Test"]`

this also works when you switch the device.

**Comment by flexsurfer** [comment]
@audriu sorry if this sounded rude or incredulously, i just wondering what is your developing process :) the problem was it should be `def` not `defn` and i have code `(if platform/ios? :white :red)` without parentheses , so i found simplest way how to do it, and decided to use it, your solution is good, but for `defn` , but i need `def` thank you for your effort

**Pull request merged**